From 709cfec2fbccc5a0b791528d4f0ad5c438edd9df Fri, 20 Oct 2017 12:02:03 +0200
From: Andrew Branson <andrew.branson@jollamobile.com>
Date: Fri, 20 Oct 2017 11:44:56 +0200
Subject: [PATCH] Revert "Move away of deprecated gnome-common"

We're not up-to-date enough with Gnome builds to support this yet.
This reverts commit 4301ab58efc217409c588a5527f68990b4e3d220.

diff --git a/Makefile.am b/Makefile.am
index 07d2bdc..5bd01f0 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,5 +1,5 @@
 include $(top_srcdir)/Makefile.decl
-ACLOCAL_AMFLAGS = --install -I m4
+ACLOCAL_AMFLAGS = -I m4
 AM_DISTCHECK_CONFIGURE_FLAGS = --enable-introspection=no --enable-vala
 
 if ENABLE_DOC
diff --git a/autogen.sh b/autogen.sh
index 2bd7c50..42ca34c 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -3,35 +3,14 @@
 srcdir=`dirname $0`
 test -z "$srcdir" && srcdir=.
 
+ORIGDIR=`pwd`
+cd $srcdir
 
-(test -f $srcdir/configure.ac) || {
-	echo "**Error**: Directory "\`$srcdir\'" does not look like the top-level project directory"
-	exit 1
-}
+# Automake requires that ChangeLog exists.
+touch ChangeLog
 
-PKG_NAME=`autoconf --trace 'AC_INIT:$1' "$srcdir/configure.ac"`
-
-if [ "$#" = 0 -a "x$NOCONFIGURE" = "x" ]; then
-	echo "**Warning**: I am going to run \`configure' with no arguments." >&2
-	echo "If you wish to pass any to it, please specify them on the" >&2
-	echo \`$0\'" command line." >&2
-	echo "" >&2
-fi
-
-set -x
-aclocal --install || exit 1
-autoreconf --verbose --force --install -Wno-portability || exit 1
-set +x
-
-if [ "$NOCONFIGURE" = "" ]; then
-        set -x
-        $srcdir/configure "$@" || exit 1
-        set +x
-
-        if [ "$1" = "--help" ]; then exit 0 else
-                echo "Now type \`make\' to compile $PKG_NAME" || exit 1
-        fi
-else
-        echo "Skipping configure process."
-fi
+REQUIRED_M4MACROS=introspection.m4 \
+	REQUIRED_AUTOMAKE_VERSION=1.11 \
+	gnome-autogen.sh "$@" || exit 1
+cd $ORIGDIR || exit $?
 
diff --git a/configure.ac b/configure.ac
index 00e50f0..9d2fc27 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2,15 +2,14 @@
 AC_CONFIG_SRCDIR([Makefile.am])
 AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_MACRO_DIR([m4])
-AM_INIT_AUTOMAKE([1.11 foreign check-news dist-xz no-dist-gzip tar-ustar])
+AM_INIT_AUTOMAKE([1.11 check-news dist-xz no-dist-gzip tar-ustar])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
 # Checks for programs.
 AC_PROG_CC
 AM_PROG_CC_C_O
-
-LT_PREREQ([2.2.6])
-LT_INIT([disable-static])
+AC_DISABLE_STATIC
+AC_PROG_LIBTOOL
 
 LIBGEE_LT_VERSION="8:0:6"
 AC_SUBST(LIBGEE_LT_VERSION)
@@ -65,7 +64,6 @@
               enable_consistency_check=$enableval, enable_consistency_check=no)
 AS_IF([test "x$enable_consistency_check" != xno], [VALA_ADD_VALAFLAGS(-D CONSISTENCY_CHECKS)])
 
-AX_REQUIRE_DEFINED([GOBJECT_INTROSPECTION_CHECK])
 GOBJECT_INTROSPECTION_CHECK([0.9.0])
 
 VALA_CHECK([0.25.1])
diff --git a/m4/vala.m4 b/m4/vala.m4
new file mode 100644
index 0000000..96fca8a
--- /dev/null
+++ b/m4/vala.m4
@@ -0,0 +1,48 @@
+AC_DEFUN([VALA_ADD_CHECKFILE],
+[
+    vala_checkfiles="$vala_checkfiles $srcdir/$1"
+])
+
+AC_DEFUN([VALA_ADD_VALAFLAGS],
+[
+    VALAFLAGS="${VALAFLAGS:+$VALAFLAGS }$1"
+])
+
+AC_DEFUN([VALA_CHECK],
+[
+    AC_ARG_ENABLE([vala],
+        [AS_HELP_STRING([--enable-vala],[enable checks for vala])],,
+            [enable_vala=no])
+    AC_ARG_ENABLE([vala-fatal-warnings],
+        [AS_HELP_STRING([--enabla-vala-fatal-warnings],[Treat vala warnings as fatal])],,
+            [enable_vala_fatal_warnings=no])
+    AS_IF([test "x$enable_vala_fatal_warnings" = "xyes"],
+          [VALA_ADD_VALAFLAGS([--fatal-warnings])])
+    AC_SUBST([VALAFLAGS])
+    dnl Enable check for Vala even if not asked to do so if checkfile files are absent.
+    for checkfile in $vala_checkfiles
+    do
+        AS_IF([test ! -e "$checkfile"],
+              [AC_MSG_WARN([Missing checkfile file $[]checkfile. Forcing vala mode])
+               enable_vala=yes
+              ])
+    done
+
+    dnl Vala
+    AS_IF([test x$enable_vala = xyes],
+          [
+           dnl check for vala
+           AM_PROG_VALAC([$1])
+           AS_IF([test x$VALAC = "x"],
+                 [AC_MSG_ERROR([Cannot find the "valac" compiler in your PATH])],
+                 [VALA_CHECK_PACKAGES([$2])])
+
+           dnl check for vapigen
+           AC_PATH_PROG(VAPIGEN, vapigen, no)
+           AS_IF([test x$VAPIGEN = "xno"],
+                 [AC_MSG_ERROR([Cannot find the "vapigen compiler in your PATH])])
+
+           ],
+           []
+    )
+])
diff --git a/m4/valacheck.m4 b/m4/valacheck.m4
deleted file mode 100644
index 96fca8a..0000000
--- a/m4/valacheck.m4
+++ /dev/null
@@ -1,48 +0,0 @@
-AC_DEFUN([VALA_ADD_CHECKFILE],
-[
-    vala_checkfiles="$vala_checkfiles $srcdir/$1"
-])
-
-AC_DEFUN([VALA_ADD_VALAFLAGS],
-[
-    VALAFLAGS="${VALAFLAGS:+$VALAFLAGS }$1"
-])
-
-AC_DEFUN([VALA_CHECK],
-[
-    AC_ARG_ENABLE([vala],
-        [AS_HELP_STRING([--enable-vala],[enable checks for vala])],,
-            [enable_vala=no])
-    AC_ARG_ENABLE([vala-fatal-warnings],
-        [AS_HELP_STRING([--enabla-vala-fatal-warnings],[Treat vala warnings as fatal])],,
-            [enable_vala_fatal_warnings=no])
-    AS_IF([test "x$enable_vala_fatal_warnings" = "xyes"],
-          [VALA_ADD_VALAFLAGS([--fatal-warnings])])
-    AC_SUBST([VALAFLAGS])
-    dnl Enable check for Vala even if not asked to do so if checkfile files are absent.
-    for checkfile in $vala_checkfiles
-    do
-        AS_IF([test ! -e "$checkfile"],
-              [AC_MSG_WARN([Missing checkfile file $[]checkfile. Forcing vala mode])
-               enable_vala=yes
-              ])
-    done
-
-    dnl Vala
-    AS_IF([test x$enable_vala = xyes],
-          [
-           dnl check for vala
-           AM_PROG_VALAC([$1])
-           AS_IF([test x$VALAC = "x"],
-                 [AC_MSG_ERROR([Cannot find the "valac" compiler in your PATH])],
-                 [VALA_CHECK_PACKAGES([$2])])
-
-           dnl check for vapigen
-           AC_PATH_PROG(VAPIGEN, vapigen, no)
-           AS_IF([test x$VAPIGEN = "xno"],
-                 [AC_MSG_ERROR([Cannot find the "vapigen compiler in your PATH])])
-
-           ],
-           []
-    )
-])

